### PR TITLE
HID-1439 update urls when filtering users, plus refactoring and tests

### PR DESCRIPTION
**Pull request by amyvbenson**

### CODE REVIEW

File: src/app/components/user/user-filters.html [hunk 1]
```diff
@@ -37,9 +32,9 @@ <h2 class="sidebar__heading">Filter</h2>
         <div class="filter" ng-if="!list || list.type !== 'disaster'">
           <label for="disasters" class="sr-only" translate>Filter by disaster</label>
           <ui-select ng-model="selectedFilters['disasters.list']" name="disasters">
-            <ui-select-match allow-clear="true" placeholder="{{'Disaster' | translate}}">{{$select.selected.name}}</ui-select-match>
+            <ui-select-match allow-clear="true" placeholder="{{'Disaster' | translate}}">{{$select.selected.label}}</ui-select-match>
             <ui-select-choices refresh="getDisasters($select.search)" repeat="disaster._id as disaster in disasters | filter: $select.search" minimum-input-length="3">
-              {{disaster.name}}
+              {{disaster.label}}
```
Comment (guillaumev) [inline src/app/components/user/user-filters.html]: Why did you change from name to label ? The "name" attribute coming from the API is supposed to be the one to use. This is especially important for organizations where name = label + acronym.

Comment (amyvbenson) [reply src/app/components/user/user-filters.html]: It was just to make it consistent with the other ones, but if the name is what should be used for disasters I'll change it back.

Comment (guillaumev) [reply src/app/components/user/user-filters.html]: Actually name is what should be used for all lists, not only disasters...
